strawberryfields/decompositions.py

```diff
@@ -136,24 +138,66 @@ def graph_embed(A, max_mean_photon=1.0, make_traceless=True, tol=1e-6):
     if m != n:
         raise ValueError("The matrix is not square.")
 
-    if np.linalg.norm(A-np.transpose(A)) >= tol:
+    if not np.allclose(A, np.transpose(A), rtol=rtol, atol=atol):
         raise ValueError("The matrix is not symmetric.")
 
     if make_traceless:
-        A = A - np.trace(A)*np.identity(n)/n
+        A = A - np.trace(A) * np.identity(n) / n
 
-    s, U = takagi(A, tol=tol)
-    sc = np.sqrt(1.0+1.0/max_mean_photon)
-    vals = -np.arctanh(s/(s[0]*sc))
+    s, U = takagi(A, tol=atol)
+    sc = np.sqrt(1.0 + 1.0 / max_mean_photon)
+    vals = -np.arctanh(s / (s[0] * sc))
+    return vals, U
+
+
+def graph_embed(A, mean_photon=1.0, make_traceless=False, atol=1e-08):
+    r"""Embed a graph into a Gaussian state.
+
+    Given a graph in terms of a symmetric adjacency matrix
+    (in general with arbitrary complex off-diagonal and real diagonal entries),
+    returns the squeezing parameters and interferometer necessary for
+    creating the Gaussian state whose off-diagonal parts are proportional to that matrix.
```

Comment:
How to the on-diagonal parts of the Gaussian state look in terms of the matrix, are they independent? Does this also depend on `make_traceless`?

Reply:
There is a connection but the connection is far from trivial. In any case, if you make traceless the adjacency matrix you *do* change the covariance matrix of the Gaussian state